### ISSUE TITLE
Improve port selection during testing

### ISSUE THREAD

**Issue by twoseat** [cloudfoundry/cf-java-client#863]
We're seeing intermittent test failures due to port clashes (e.g. https://java-experience.ci.springapps.io/teams/java-experience/pipelines/java-client/jobs/integration-test-2.0-release-master/builds/4). They relate to a hard-coded port we use, which may be the wrong decision, or we may need to find a way to actively identify a free port.

**Comment by nebhale** [comment]
Yeah, that’s certainly a problem.  I’m guessing that you’ll need to query the CC API to determine what ports are already reserved and choose something else?

**Comment by twoseat** [comment]
That's my first guess. There is an option to have CC pick a port, but that may defeat the point of the offending test.
